File: src/main/java/io/lcydev/spawnercleaner/listeners/SpawnerListener.java
package io.lcydev.spawnercleaner.listeners;

import io.lcydev.spawnercleaner.SpawnerCleaner;
import io.lcydev.spawnercleaner.utils.Config;
import org.bukkit.Location;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.SpawnerSpawnEvent;

public class SpawnerListener implements Listener {

    private final SpawnerCleaner instance;
    private final Config config;

    public SpawnerListener() {
        instance = SpawnerCleaner.inst();
        config = Config.get();
    }

    @EventHandler
    public void onSpawnerEvent(SpawnerSpawnEvent e) {
        if (!config.getWorlds().contains(e.getLocation().getWorld().getName())) return;

        Location spawnerLoc = e.getSpawner().getLocation();
        instance.removeQueue.add(spawnerLoc);
        e.setCancelled(true);
    }
}

File: src/main/kotlin/io/lcydev/spawnercleaner/listeners/SpawnerListener.kt
package io.lcydev.template.listeners

import io.lcydev.template.SpawnerCleaner
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener
import org.bukkit.event.entity.SpawnerSpawnEvent

class SpawnerListener(private val plugin: SpawnerCleaner) : Listener {

    @EventHandler
    fun onSpawnerEvent(e: SpawnerSpawnEvent) {

        val spawner = e.spawner
        val location = spawner?.location
        val world = location?.world

        if (false) return

        if (spawner != null) {
            plugin.removeQueue.add(location)
        }
        e.isCancelled = true
    }
}